File: PHP/kode/MusicInstrument.php
<?php
require_once 'Item.php';

class MusicInstrument extends Item {
    private $playingMethod=''; // dipukul / ditabuh / dipetik
    private $condition='';     // likenew / baru / bekas / ori / premium
    private $weight='';        // disimpan sebagai string supaya satuan "kg" ikut

    public function __construct($id = 0, $price = 0.0, $stock = 0,
                                 $playingMethod = "", $condition = "", $weight = "", $photo = "") {
        parent::__construct($id, $price, $stock, $photo);
        $this->playingMethod = $playingMethod;
        $this->condition = $condition;
        $this->weight = $weight;
    }

    // getter and setter playingMethod
    public function setPlayingMethod($playingMethod) { $this->playingMethod = $playingMethod; }
    public function getPlayingMethod() { return $this->playingMethod; }

    // getter and setter condition
    public function setCondition($condition) { $this->condition = $condition; }
    public function getCondition() { return $this->condition; }

    // getter and setter weight
    public function setWeight($weight) { $this->weight = $weight; }
    public function getWeight() { return $this->weight; }
}
?>

File: Cpp/Kode/MusicInstrument.cpp
#include "Item.cpp"

class MusicInstrument : public Item{
    private :
        string playingMethod; // dipukul / ditabuh / dipetik
        string condition; // likenew / baru / bekas / ori / premium
        string weight; // weight disimpan sebagai string supaya ikut satuan "kg"
    public :
        // constructor kosong
        MusicInstrument(){
            playingMethod = "";
            condition = "";
            weight = "";
        };
        // constructor berparameter (memanggil constructor Item lewat inisialisasi list)
        MusicInstrument(int id, double price, int stock,
                         string playingMethod, string condition, string weight)
            : Item(id, price, stock){
            this->playingMethod = playingMethod;
            this->condition = condition;
            this->weight = weight;
        }
        // getter and setter playingMethod
        void setPlayingMethod(string playingMethod){this->playingMethod=playingMethod;}
        string getPlayingMethod(){return playingMethod;}

        // getter and setter condition
        void setCondition(string condition){this->condition=condition;}
        string getCondition(){return condition;}

        // getter and setter weight
        void setWeight(string weight){this->weight=weight;}
        string getWeight(){return weight;}

        // destructor
        virtual ~MusicInstrument(){}
};